### ISSUE TITLE
Piston's website is deceptive

### ISSUE THREAD

**Issue by icefoxen** [PistonDevelopers/piston#1182]
It describes Piston as "a modular, open source game engine", which implies a coherent system with a defined goal.  In contrast, Piston appears to be a name that gets applied to the project of anyone who wants it, even when it is only peripherally related to games (eg, Dyon, eco), is basically trivial (rect), or has better solutions (float, quaternion vs. num, nalgebra).

More accurately describing the goals of Piston and what to expect from it would be desirable.

**Comment by bvssvni** [comment]
No, the term "modular" means modular. Piston is modular by design. The goals are clearly defined in the README here https://github.com/PistonDevelopers/piston#goals. What you are confused about is the modular core of Piston and project Piston in general (the game engine). You can find some answers of your questions in the [FAQ](https://github.com/PistonDevelopers/piston/wiki/Frequently-Asked-Questions-(FAQ)).

- The Piston project is the umbrella which aims to develop a full feature game engine. Because this requires a lot of components, they are being worked on people in parallel. We try to decouple projects and even separate out projects when they grow big enough to serve a standalone feedback from a community with diversified interests. Piston could have started its own 3D API, but we discussed this and went with supporting Gfx, such that we could focus on more important things. Some components, like the Image library, was not available in Rust so we worked on it. Not everybody are interested in working on a game engine, but they need image encoding and decoding so they contribute to the project. Because of the modular design of Piston, nobody needs to worry about polling in dependencies from the game engine they don't need.
- A game engine is what makes you productive when building games. In order to build a game engine you need tools to maintain the ecosystem efficiently. That is why we developed Eco and moved VisualRust here when maintainers were needed.
- Developing a game engine in a new programming language is hard, that is why we need as much experimentation as possible. Tools that makes it easier to experiment, e.g. Dyon is both fun to work on and gives important feedback since we learn how scripting languages interfaces with the engine. Dyon is part of the Piston project, but it of course not part of the modular core. It can be used as standalone library, but it is a component that can be used as part of a game engine. We are not trying to develop a monolithic game engine, but use a modular architecture because the development goes over a long period of time, and this makes it possible to bits and pieces and test them in real world applications.
- Some libraries start out as an experiment and slows down in development. This is natural. Things change over time.
- The math libraries in Piston are designed for compatibility across projects and reexporting. A standard pattern in Piston making a `math` module in a library with helper functions specifically designed for that particular purpose. The reason for this is that no math library cover all needs. Mathematics is used everywhere, but the basic matrix functions are reexported from a common library to avoid redundant code.

Can you elaborate on what you mean about "what to expect from it"? I think you have something important to say here.

**Comment by bvssvni** [comment]
I'm closing this since I believe it was based on a misunderstanding. Please reopen if there is a problem.
